pkg/api/upgradelog/handler.go

```diff
@@ -33,13 +33,38 @@ const (
 set -e
 
 echo "clean up stale archives, if any"
-
 rm -vf /archive/*.zip
 
 echo "start to package upgrade logs"
 
+tmpdir=$(mktemp -d)
+mkdir $tmpdir/logs
+
+# We used to collect logs in plain text until v1.2.0, the packaging script
+# should be able to determine what kind of post-processing to execute.
+# Starting from v1.2.0, the upgrade logs will be collected in .gzip format.
+is_plain_log="false"
+
 cd /archive
-zip -r $ARCHIVE_NAME ./logs/
+for f in logs/*.log
+do
+    is_plain_log="true"
```

Comment:
few questions:
1. `is_plain_log` has leading space instead of tab, comparing to other lines
2. `is_plain_log` is set to `true` anyway in the do loop, is it wrong ?
3. L61 assumes that the *.log is either plain text, or in given format, there is no possibility that they are mixed,  then,  the code could be optimized as:

(1) check if any file is in `is_plain_log`  format, ant set the variable, break;  or simply check the first file to determin the potential format;  or simply use command like `file` to check the format of of
```
file sidebars.js 
sidebars.js: ASCII text

file sb_e040_20220510.zip 
sb_e040_20220510.zip: Zip archive data, at least v1.0 to extract

```
(2) if `is_plain_log`, then copy all file to the tmpdir

L50 do loop all log files with `awk` and `jq`, not necessary

Reply:
1. Good spot, thanks.
2. It is a stupid way of doing. I've changed the way we detect the format.
3. Same as above.

Thanks for the suggestion!

Reply:
Sadly, we're not able to bring the `file` command into v1.1.2's `harvester-upgrade` image, so I'll just fallback to the original way.